crates/ra_hir/src/name.rs

```diff
@@ -184,3 +186,12 @@ impl AsName for KnownName {
         Name::new(s.into())
     }
 }
+
+fn resolve_name(text: SmolStr) -> SmolStr {
+    let raw_start = "r#";
+    if text.as_str().starts_with(raw_start) {
+        SmolStr::new(&text[raw_start.len()..])
+    } else {
+        text
+    }
+}
```

Comment:
Two nits:

* it seems cleaner to accept `&SmolStr` and invoke `.clone`, rather than just drop an owned value in case of `r#`
* let's name this `cook_name`: that's what you do to process raw things

Reply:
Fixed